sbpy/spectroscopy/core.py

```diff
@@ -9,29 +9,270 @@
+
 
 __all__ = ['Spectrum', 'SpectralModel']
 
 
-def einstein_coeff(frequency):
+def constants(temp_estimate, transition_freq, mol_tag, vgas):
+
+    """
+    Returns relevant constants from JPLSpec catalog and energy calculations
+
+    Parameters
+    ----------
+    transition_freq : `~astropy.units.Quantity`
+        Transition frequency in MHz
+
+    temp_estimate : `~astropy.units.Quantity`
+        Estimated temperature in Kelvins
+
+    mol_tag : int or str
+        Molecule identifier. Make sure it is an exclusive identifier.
+
+    vgas : `~astropy.units.Quantity`
+        Gas velocity approximation in km / s
+
+    Returns
+    -------
+    constants : list
+        List of constants in the following order:
+            | Transtion frequency
+            | Temperature
+            | Integrated line intensity at 300 K
+            | Partition function at 300 K
+            | Partition function at designated temperature
+            | Upper state degeneracy
+            | Upper level energy in Joules
+            | Lower level energy in Joules
+            | Degrees of freedom
+            | Planck Constant
+            | Speed of light
+            | Boltzmann constant
+
+    """
+
+    query = JPLSpec.query_lines(min_frequency=(transition_freq - (100 * u.MHz)),
+                                max_frequency=(transition_freq + (100 * u.MHz)),
+                                molecule=mol_tag)
+
+    freq_list = query['FREQ']
+
+    t_freq = min(list(freq_list.quantity), key=lambda x: abs(x-transition_freq))
+
+    data = query[query['FREQ'] == t_freq.value]
+
+    df = int(data['DR'].data)
+
+    lgint = float(data['LGINT'].data)
+
+    lgint = 10**lgint * u.nm * u.nm * u.MHz
+
+    elo = float(data['ELO'].data) / u.cm
+
+    gu = float(data['GUP'].data)
+
+    cat = JPLSpec.get_species_table()
+
+    mol = cat[cat['TAG'] == mol_tag]
+
+    temp_list = cat.meta['Temperature (K)'] * u.K
+
+    temp = min(temp_list, key=lambda x: abs(x-temp_estimate))
+
+    for i in range(1, 8):
+        if mol.columns['QLOG{}'.format(i)].meta !=\
+           {'Temperature (K)': temp.value}:
+
+            pass
+
+        else:
+            partition = 10 ** (float(mol.columns['QLOG{}'.format(i)].data))
+
+    part300 = 10 ** (float(mol['QLOG1'].data))
+
+    h = con.h.to('J*s')  # planck constant
+
+    k = con.k_B.to('J/K')  # boltzman constant
+
+    c = con.c.to('m/s')  # speed of light
+
+    vgas = vgas.to('m/s')  # convert our vgas to m/s
+
+    # yields in 1/cm
+    energy = elo + (t_freq.to(1/u.cm, equivalencies=u.spectral()))
+
+    scale = gu*np.exp(-energy.to(u.J, equivalencies=u.spectral()) /
+                      (k*temp))/partition  # upper level energy
+
+    energy_J = energy.to(u.J, equivalencies=u.spectral())
+    elo_J = elo.to(u.J, equivalencies=u.spectral())
+
+    result = []
+
+    result.append(t_freq)
+
+    result.extend((temp, lgint, part300, partition, gu, energy_J,
+                   elo_J, df, h, c, k))
```

Comment:
You don't need to return physical constans in this function, they are available in `astropy.constants`